### PR TITLE
Modify code for obstacles

### PR DESCRIPTION
**Pull request by Zhehui-Huang**

### CODE REVIEW

File: gym_art/quadrotor_multi/quadrotor_multi.py [hunk 1]
```diff
@@ -430,34 +420,32 @@ def step(self, actions):
...
             rew_obst_quad_collisions_raw = np.zeros(self.num_agents)
             if obst_quad_last_step_unique_collisions.any():
                 # We assign penalties to the drones which collide with the obstacles
-                # In obst_quad_last_step_unique_collisions, 2*i is quad_id, 2*i+1 is obstacle_id, i >= 0
-                # and we only penalize drones
-                for i in range(len(obst_quad_last_step_unique_collisions) // 2):
-                    quad_id = int(2 * i)
-                    rew_obst_quad_collisions_raw[quad_id] = -1.0
+                # And obst_quad_last_step_unique_collisions only include drones' id
+                rew_obst_quad_collisions_raw[obst_quad_last_step_unique_collisions] = -1.0
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_multi.py:433]: I see the logic was changed here.
Before you calculated quad_id as int(2*i) and not you use `obst_quad_last_step_unique_collisions` directly.
I trust you that you tested this, but this probably means that we either had a bug before, or we have one now, right?

Comment (alex-petrenko) [reply gym_art/quadrotor_multi/quadrotor_multi.py:433]: Is this the place where you fixed a bug you were mentioning?

File: gym_art/quadrotor_multi/quadrotor_multi.py [hunk 2]
```diff
@@ -510,56 +498,43 @@ def step(self, actions):
...
+                self.quads_formation_size = self.scenario.formation_size
+                self.goal_central = np.mean(self.scenario.goals, axis=0)
+                tmp_obs = self.multi_obstacles.reset(
+                    obs=obs, quads_pos=self.pos, quads_vel=quads_vel, set_obstacles=self.set_obstacles,
+                    formation_size=self.quads_formation_size, goal_central=self.goal_central)
+
+                # In testing mode, which means reset the scene, and this in visualization would make people feel
+                # the screen stuck around one second
+                if self.obstacle_num > 1:
+                    self.reset_scene = True
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_multi.py:537]: Why do we need to reset the scene here?

**Review by alex-petrenko — APPROVED** [review]

File: gym_art/quadrotor_multi/quadrotor_multi.py (hunk 1 above)
Comment (Zhehui-Huang) [reply gym_art/quadrotor_multi/quadrotor_multi.py:433]: We had a bug before, and then I use an extra array to only save the drone id for calculating the penalty, which means obst_quad_last_step_unique_collisions only include the drone id.

Comment (Zhehui-Huang) [reply gym_art/quadrotor_multi/quadrotor_multi.py:433]: > Is this the place where you fixed a bug you were mentioning?

Yes

File: gym_art/quadrotor_multi/quadrotor_multi.py (hunk 2 above)
Comment (Zhehui-Huang) [reply gym_art/quadrotor_multi/quadrotor_multi.py:537]: At first, I was designed to make our repository support for multiple different shapes of obstacles, such as cube and sphere. Therefore, every time when we reset obstacles, we should also update the shape of obstacles in visualization(or render). And use `self.reset_scene = True` is the most easiest way. Although we can design a way to make it more efficient, but this just like hacky solution.

**Pull request merged**